modules/add_credit/src/Plugin/Field/FieldWidget/UnitPriceRangeDefaultWidget.php

```diff
@@ -0,0 +1,149 @@
+<?php
```

Comment:
This widget pretty much supplies the same form as `commerce_unit_price`. I am wondering if it would be better to add a custom constraint to the `unit_price` field in `hook_field_info` that handles the validation.

Reply:
This is what I had before. But there was a lot of overrides: defautSettings, settingsForm, formElement, extractFormValues..etc. All are handled differently. So it was much easier to extend WidgetBase.

Reply:
Great idea! I've implemented this. 👍

Reply:
We don't need the new widget anymore and top up product type can use the default order item.